website/docs/r/token.html.markdown

```diff
@@ -0,0 +1,43 @@
+---
+layout: "sumologic"
+page_title: "SumoLogic: sumologic_token"
+description: |-
+  Provides a Sumologic Token
+---
+
+# sumologic_token
+Provides a [Sumologic Token][1].
+
+## Example Usage
+```hcl
+resource "sumologic_token" "example_token" {
+  name          = "testToken"
+  description   = "Testing resource sumologic_token"
+  status        = "Active"
+  type          = "CollectorRegistration"
+  version       = 0
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+  * `name` - (Required) Display name of the token. This must be unique across all of the tokens.
+  * `description` - (Optional) The description of the token.
+  * `status` - (Required) Status of the token. Supported values are `Active`, and `Inactive`.
+  * `type` - (Required) Type of the token. Supported value is `CollectorRegistration`.
+  * `version` - (Required for update) Version of the token. This is only required for update. It starts from 0 when created and gets incremented with each update.
```

Comment:
not TF specific question: why do we require user to manage the version?

Reply:
As for `version`, this is required by content library framework.
It's not needed for `CREATE` since framework will have all objects with version starting from 0. The version is required for `UPDATE`. The reason is that we need to make sure they don't have inconsistent update. It works similarly to `Etag` in collector update request.

Reply:
yeah I understand it's to prevent race conditions on update, yet we don't require user to specify an `etag` in terraform for sources do we?

Reply:
It's a good question. I dig into the code and it seems that it doesn't mention about the `ETag` anywhere in collector/source terraform. It seems that `ETag` is handled in [api.go](/Users/yuting/git/terraform-provider-sumologic/vendor/github.com/aws/aws-sdk-go/service/s3/api.go)

The current `required` for version aligns with our token [api doc](https://nite-api.sumologic.net/docs/#operation/updateToken). 

Want to have confirmation from TF SME on this @vsinghal13 @seansain

Reply:
Ping @vsinghal13 @seansain @sumovishal again on this issue.

Reply:
From looking at the Monitors resource there is also a `version` parameter, but it is not documented. I will do some digging to check how it works.

Reply:
We can follow the same precedence as monitors here. We do not need to expose `version` in the docs. It is updated by the server and the right value is returned back to us so Terraform will store that. So, when the user tries to update the resource they will have the right `version`.